pkg/api/validation/validation.go

```diff
@@ -1481,7 +1483,15 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(specPath.Child("storageClassName"), pv.Spec.StorageClassName, msg))
 		}
 	}
-
+	if pv.Spec.VolumeMode != nil && !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
```

Comment:
Ok, I think the problem is that we default the field to Filesystem, so it is always going to be set by the time we reach validation.  We should either
1. not default it, and have the code handle a nil value, or
2. check for the feature gate if the value is Block.

Reply:
we only default the pvc not the pv

Reply:
I would combine the feature gate check and the volume mode values check under one block.